### PR TITLE
Cryoxadone works while conscious and is affected by purity

### PR DESCRIPTION
**Pull request by RikuTheKiller**
## About The Pull Request

It no longer straight up denies healing based on if you're knocked out or not, but rather, it just becomes less effective.

Cryoxadone is now also affected by purity and is only 75% pure by default. (standard purity)

Clearing up a previous misconception, my changes actually made cryoxadone slower, not faster. I've remedied that fact and cryoxadone in cryotubes should stay the same as it was before.

100% pure cryoxadone while conscious has the same exact speed as 75% pure cryoxadone while sleeping. 100% cryoxadone while sleeping is the fastest. In other words, grabbing roundstart cryoxadone and stabbing it into yourself won't be nearly as fast (partly due to the lack of extremely low temperatures) as just using it in a cryotube.

## Why It's Good For The Game

Cryoxadone never really had the same effect here as it did on goon due to how slowly it heals and how many disadvantages being cold gives. There was never any reason to limit cryoxadone to unconscious-only except for justifying the use of anesthetic mix in cryo tubes.

This will also finally stop medics from killing and potentially round-removing people by accidentally slamming them into a cryo tube without disabling their internals. It happens more often than you'd think.

The PR also adds consistency via making it affected by purity like most standard reagents.

Right, I was told to update some of the reasoning here. A main part of this PR is to enable cryoxadone to be used in combat once more. Making it require sleep completely denies it's use in combat. (it's still not OP due to the fact being cold slows you down significantly)
## Changelog
:cl:
balance: Cryoxadone can be used while conscious again and is affected by purity.
/:cl:

### CODE REVIEW

**Comment by Pickle-Coding** [comment]
I feel like people could use this to cheese breaches, and that would be sad.

Getting cryod does pretty much everything automatically except healing organ damage. Being unconcious is a good requirement to restrain this chemical for medical usages that require cooperation, instead of being another ingredient in a paramedic's super giga powergame syringe they can use while running around like an ape.

**Comment by Jacquerel** [comment]
I am also concerned at the concept of Cryoxadone working at what is its current full strength while cold for any reason, rather than in a cryotube.
It used to be reasonably common to give people the chemical and cool them down without bothering with the inconvenience of the tube, I don't see any reason this behaviour would not immediately return following this change.

**Comment by ATH1909** [comment]
> I am also concerned at the concept of Cryoxadone working at what is its current full strength while cold for any reason, rather than in a cryotube. It used to be reasonably common to give people the chemical and cool them down without bothering with the inconvenience of the tube, I don't see any reason this behaviour would not immediately return following this change.

personally, I thought that that was pretty cool.

doesn't the cryo tube also multiply chems? if so, couldn't the fix for cryo showers just be to make cryoxadone metabolize faster (instead of a hard "no, use the cryo tube or it does nothing"), allowing cryo tubes to be more efficient but not near-mandatory?

**Comment by RikuTheKiller** [comment]
> > I am also concerned at the concept of Cryoxadone working at what is its current full strength while cold for any reason, rather than in a cryotube. It used to be reasonably common to give people the chemical and cool them down without bothering with the inconvenience of the tube, I don't see any reason this behaviour would not immediately return following this change.
> 
> personally, I thought that that was pretty cool.
> 
> doesn't the cryo tube also multiply chems? if so, couldn't the fix for cryo showers just be to make cryoxadone metabolize faster (instead of a hard "no, use the cryo tube or it does nothing"), allowing cryo tubes to be more efficient but not near-mandatory?

The cryotube multiplies chemicals based on it's stock parts, so yes, that is a factor.
Cryoxadone also metabolizes differently depending on temperature.

**Comment by RikuTheKiller** [comment]
> I am also concerned at the concept of Cryoxadone working at what is its current full strength while cold for any reason, rather than in a cryotube. It used to be reasonably common to give people the chemical and cool them down without bothering with the inconvenience of the tube, I don't see any reason this behaviour would not immediately return following this change.

Oh, you can do that already, just tell them to sleep emote, it's just more annoying. You can also bypass it via N2O internals or sleep reagents.

The only purpose for this PR is to make cryoxadone usable in combat once more, please try to actually use cryoxadone and see for yourselves just how slow it is even at full purity.

**Comment by RikuTheKiller** [comment]
> I feel like people could use this to cheese breaches, and that would be sad. Getting cryod does pretty much everything automatically except healing organ damage.
> 
> Being unconcious is a good requirement to restrain this chemical for medical usages that require cooperation, instead of being another ingredient in a paramedic's super giga powergame syringe they can use while running around like an ape.

There are many, many ways of getting through a breach. I don't see why cryoxadone wouldn't be one of them. Other chemicals can do the same exact thing no problem. Not to mention, I'm not even sure if roundstart cryoxadone while conscious is enough to survive a breach, it's probably around one third slower than pre-PR cryo which is already pretty slow. Space also doesn't really cool you down that far anyways.

**Comment by Pickle-Coding** [comment]
> The only purpose for this PR is to make cryoxadone usable in combat once more (even if it's slow), please try to actually use cryoxadone and see for yourselves just how slow it is even at full purity. (it's barely fast enough to be useful in combat)

That's a really terrible reason in my opinion. Cryoxadone does not look like a combat chemical. It is literally designed to work on incapacitated patients only.

We already have plenty of chemicals that can work for combat. Why are you trying to change one of the very few chemicals that don't to be combat viable?

Cryoxadone is also extremely powerful and would just be too overpowered for combat use. It heals brute, burn, toxin, oxygen, and even clone damage, simultaniously. It can heal any and all wounds too, simultaniously. It's a chemical to treat incredibly fucked up patients while doctors work on other people.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 1]
```diff
@@ -125,17 +125,18 @@
...
 	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
 
 /datum/reagent/medicine/cryoxadone/on_mob_life(mob/living/carbon/affected_mob, delta_time, times_fired)
 	metabolization_rate = REAGENTS_METABOLISM * (0.00001 * (affected_mob.bodytemperature ** 2) + 0.5)
-	if(affected_mob.bodytemperature >= T0C || !HAS_TRAIT(affected_mob, TRAIT_KNOCKEDOUT))
+	if(affected_mob.bodytemperature >= T0C)
 		..()
 		return
-	var/power = -0.00003 * (affected_mob.bodytemperature ** 2) + 3
+	var/power = (-0.00004 * (affected_mob.bodytemperature ** 2) + 3) * normalise_creation_purity() * HAS_TRAIT(affected_mob, TRAIT_KNOCKEDOUT) ? 1 : 0.75
```
Comment (Pickle-Coding) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:139]: Changing -0.00003 to -0.00004 actually makes it slightly slower. It increases the rate which temperature reduces the healing rate.

Comment (RikuTheKiller) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm:139]: Well ain't that nice to know.

Comment (RikuTheKiller) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: I've changed it back to how it was, it should no longer be slower.

**Comment by RikuTheKiller** [comment]
Well, turns out, the PR wasn't making cryoxadone faster, it was making it slower. It also turns out that 100% pure cryoxadone while conscious has the same effect as 75% pure cryoxadone while unconscious. I've updated the PR body to reflect this.

**Comment by RikuTheKiller** [comment]
This PR goes quite well with #74136 I just realized. That would still give cryotubes an advantage as long as you upgrade them. (Since they can reach very low temperatures once more.)

**Comment by RikuTheKiller** [comment]
I just tested it in-game and 75% pure cryoxadone (the roundstart kind) isn't enough to keep you alive in a spaced room on it's own. It doesn't even keep up with the burn and brute. It may be enough to stop you from dying to suffocation, though.

**Comment by RikuTheKiller** [comment]
> Having to balance cryoxadone on both combat and medbay usages just seems overcomplicated and unnecessary. I don't see the point. We already have a lot of chemicals useful for combat.
> 
> Cryoxadone has one purpose, and that is to assist really fucked up patients while doctors focus on other people. It heals every damage type and treats all wounds, all simultaniously.

You do know it's a fairly recent change that cryoxadone can't be used outside of cryotubes, right? It wasn't off-balance and not to mention it's had quite a few side-effects and kills people every now and then via being left in cryo with internals on. (Which is plain annoying.)

The reason cryoxadone is already balanced is that it's not actually THAT useful in combat. You need a large amount of it for it to be useful, especially if you compare it to something like libital or lenturi.

I asked about bringing it back in the coder channel and the people who talked with me never seemed to encounter any balancing issues with cryoxadone as it was before. We don't need to needlessly remove a purpose from it, not to mention the reason it was made this way is still just so that cryotubes would use gas.

And you keep stressing "simultaneously" but it's not like you encounter all damage types at once. Cryoxadone is incredibly slow at dealing with single damage types. It's a specific-case medicine, just like all the others.

**Comment by RikuTheKiller** [comment]
Honestly I'm sort of tired of trying to answer theoretical problems that never existed in the first place. It wasn't a problem before, why would a problem suddenly appear out of thin air? Have you ever encountered this problem in-game or thought of using cryoxadone over other options yourself?

**Comment by RikuTheKiller** [comment]
> doesn't the cryo tube also multiply chems? if so, couldn't the fix for cryo showers just be to make cryoxadone metabolize faster (instead of a hard "no, use the cryo tube or it does nothing"), allowing cryo tubes to be more efficient but not near-mandatory?

Also, ath has a very good point. Cryoxadone metabolizes extremely quickly, that's why it works so well with the multiplication in cryotubes as it provides a practically infinite source of it. It's metabolization rate is part of it's balance.

**Comment by GuillaumePrata** [comment]
Doctor malpractice adds to a round, either accidental or deliberate by an antagonist and teaching new doctors about the minute details of treatments and their dangers is one of the best sides of medbay.

The issues with healing are not theoretical, we always had issues with minmaxing multiple chems at once to get the most healing and sadly, chem purity that could fix this issue forever never got traction.

Cryoxadone by itself is not that big of a deal, but Cryoxadone has an amazing design now together with cryotubes and reverting it back to another safe chem to the minmax pile together with removing the possible malpractice is sad.

**Comment by RikuTheKiller** [comment]
> Doctor malpractice adds to a round, either accidental or deliberate by an antagonist, and teaching new doctors about the minute details of treatments and their dangers is one of the best sides of medbay.
> 
> The issues with healing are not theoretical, we always had issues with minmaxing multiple chems at once to get the most healing and sadly, chem purity that could fix this issue forever never got traction.
> 
> Cryoxadone by itself is not that big of a deal, but Cryoxadone has an amazing design now together with cryotubes and reverting it back to another safe chem to the minmax pile together with removing the possible malpractice is sad.

I'm fine with some malpractice, but cryotube fails are probably the most annoying form of malpractice I know of. It's the same as if you never got dragged to medbay in the first place. Actually, no, it's worse, because you have to consciously ghost to even get back into the round as you can't die easily which will completely prevent you from being revived.

Also it was never a safe chem to minmax. Being cold has a whole bunch of downsides, in fact, it's probably the riskiest chem to minmax even if it comes back. (apart from pyroxadone, which is riskier) Though... hard-capping things has always just felt bad for me, so I'd like it if we got some kind of consistent way of stopping it without it being unfun or unreasonable.

In fact, I sort of hate chem purity due to how finnicky it is. Most people don't care about it and trying to care about it makes you slowly go insane as 1 variable changes on a chem and now it cascades into having to change like 7 recipes and never getting time to refine them fully. Purity is pain and it's not really a necessary evil, but until someone steps up to the plate to make another fix, we're sort of stuck with it.

**Comment by RikuTheKiller** [comment]
> Personally, I throw out cryoxodone from the tubes. Not even joking, it’s usually better to make a soup of other healing chems and slap it in a cryotube to heal someone than it is to use straight cryoxodone. even does the victims a favor and makes sure they get healed regardless, even if that healing comes with side effects. Also, I will stress this one time and one time only: Incompetent cryo is _**NOT**_ fun under any circumstance, and can end up essentially being a form of round removal (especially on lowpop or in situations where medbay gets Massacred. it happens more often than I like.)

Thank you, I was starting to think I was alone on being frustrated by incompetent cryo. It's plain infuriating and I've probably experienced it on both goon and here around 20 times total. It's fucking hell, man.

But yeah, other chems are indeed better, I've made that point several times. Cryoxadone is just good at being a broad medicine and that's all it is. It will never beat other, more specific chems in combat, because it was never intended to do that in the first place. Though, I'd be happy to make it able to do that if you bothered to go out of your way to upgrade the thermomachine and all that.

**Comment by GuillaumePrata** [comment]
Cryoxadone is an amazing healing for combat because it heals wounds, something that was directly designed to force people to go into medbay for healing or do some ghetto medicine with side effects.

You don't need to be cold all the time to have Cryo as a powerful combat healing mix, just need to spray yourself with Hercuri and suffer a little slowdown until your wounds get healed.

**Comment by RikuTheKiller** [comment]
Anyways with that back and forth out of the way, we should probably make a final decision on whether to merge this or not. Points have been made, countered and remade.

**Pull request closed without merging**